trigger/src/main/scala/core/scheduler/executors/spark/SparkExecutor.scala

```diff
@@ -0,0 +1,85 @@
+import akka.stream.ActorMaterializer
+
+import scala.collection.JavaConverters._
+import org.apache.spark.launcher.SparkLauncher
+import play.api.libs.json.{JsValue, Json}
+import play.api.libs.ws.ahc.StandaloneAhcWSClient
+import core.models.enums.JobStatuses._
+import core.scheduler.executors.Executor
+import core.scheduler.utilities.SparkExecutorConfig
+import play.api.libs.ws.JsonBodyReadables._
+
+object SparkExecutor extends Executor {
+
+  override def execute(jobInstance: JobInstance, updateJob: JobInstance => Future[Unit])
+                      (implicit executionContext: ExecutionContext): Future[Unit] = {
+    jobInstance.executorJobId match {
+      case None => submitJob(jobInstance, updateJob)
+      case Some(executorJobId) => updateJobStatus(executorJobId, jobInstance, updateJob)
+    }
+  }
+
+  private def submitJob(jobInstance: JobInstance, updateJob: JobInstance => Future[Unit])
+                       (implicit executionContext: ExecutionContext): Future[Unit] = {
+    val id = randomUUID().toString
+    val ji = jobInstance.copy(executorJobId = Some(id), jobStatus = Submitting)
+    updateJob(ji).map { _ =>
+      val running = getSparkLauncher(id, ji.jobName).launch()
+      Thread.sleep(SparkExecutorConfig.getSubmitTimeOut)
+      running.destroyForcibly()
+    }
+  }
+
+  private def updateJobStatus(executorJobId: String, jobInstance: JobInstance, updateJob: JobInstance => Future[Unit])
+                             (implicit executionContext: ExecutionContext): Future[Unit] = {
+    val wsClient = getWsClient
+    wsClient.url(getStatusUrl(executorJobId)).get().map { response =>
+      (Json.fromJson[AppsResponse](response.body[JsValue]).asOpt match {
+        case Some(asd) => asd.apps.app
+        case None => Seq.empty
+      }) match {
+        case Seq(first) => updateJob(jobInstance.copy(jobStatus = getStatus(first.finalStatus)))
+        case _ => updateJob(jobInstance.copy(jobStatus = Lost))
+      }
+    }
+  }
+
+  private def getSparkLauncher(id: String, jobName: String): SparkLauncher = {
+    new SparkLauncher(Map("HADOOP_CONF_DIR" -> SparkExecutorConfig.getHadoopConfDir).asJava)
+      .setMaster(SparkExecutorConfig.getMaster)
+      .setDeployMode(SparkExecutorConfig.getDeploymentMode)
+      .setMainClass(SparkExecutorConfig.getMainClass)
+      .setAppResource(SparkExecutorConfig.getAppResource)
+      .setSparkHome(SparkExecutorConfig.getSparkHome)
+      .setAppName(jobName)
+      .setConf("spark.yarn.tags", id)
+      .addAppArgs()
+  }
+
+  private def getWsClient: StandaloneAhcWSClient = {
+    StandaloneAhcWSClient()(ActorMaterializer()(ActorSystem()))
+  }
+
+  private def getStatusUrl(executorJobId: String): String = {
+    s"${SparkExecutorConfig.getHadoopResourceManagerUrlBase}/ws/v1/cluster/apps?applicationTags=$executorJobId"
+  }
+
+  private def getStatus(finalStatus: String): JobStatus = {
+    finalStatus match {
+      case fs if fs == "UNDEFINED" => Running
```

Comment:
Maybe extracting the literals into a companion object or something similar?

Reply:
Yes. Fixed.